recipes/license_server.rb

```diff
@@ -25,19 +24,26 @@
 data_bag_item = bag.to_hash
 total_available_license_quota = 0
 
-license_groups = data_bag_item.inject('enterprise' => {}) do |hash, (key, value)|
+license_groups = data_bag_item.inject({}) do |hash, (key, value)|
   unless %w[id chef_type data_bag].include? key
     doc = Nokogiri::XML value
+    sourcetypes = doc.search('//sourcetype').map(&:text).join
     type = doc.at_xpath('/license/payload/type/text()').to_s
+    type = "#{type}_#{Digest::SHA256.hexdigest(sourcetypes).upcase}" if type == 'fixed-sourcetype'
     quota = doc.at_xpath('/license/payload/quota/text()').to_s.to_i
     expiration_time = doc.at_xpath('/license/payload/expiration_time/text()').to_s.to_i
-    total_available_license_quota += quota if type == 'enterprise' && expiration_time > Time.now.to_i
+    total_available_license_quota += quota if (type == 'enterprise' || type == "fixed-sourcetype_#{Digest::SHA256.hexdigest(sourcetypes).upcase}") && expiration_time > Time.now.to_i
     hash[type] ||= {}
+    fail 'Multiple license types are not currently supported' if hash.length > 1
     hash[type][key] = value
   end
+  node.run_state['type'] = type
```

Comment:
This should either move as is to right after the fail ... if on line 37 
OR
This should be outside the inject loop after the end on line 42, slightly modified like this: 
```ruby
node.run_state['type'] = license_groups.keys.first
```
That type variable defined on line 31 feels like it's really intended to only be valid within the unless block (line 28 -> 39), 

If you happen to be retrieving a level where skipping the keys 'id', 'chef_type' and 'data_bag' mattered, and one of those happened to be traversed prior to your actual key that you're interested in, you'll have a ```NameError: undefined local variable or method `type'``` on your hands

Reply:
nit: should 'type' be 'license_type' or 'stack_id' ... something a bit more descriptive as a key?

Reply:
that makes sense, these changes are reflected in faaab86